### PR TITLE
color grouping by dropping to chart implemented

### LINKED ISSUE #1946: Color group observed data when drag and dropping folders to charts
When adding data sets to a chart via drag-and-drop of a folder (or multiple folders), there should be the possibility to get them all assigned the same color in the chart. The whole feature specification can be found in the corresponding issue in Core:

https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/issues/1337

### PR DESCRIPTION
**Pull request by georgeDaskalakis**
close #1946

### CODE REVIEW

File: src/PKSim.Assets/PKSimConstants.cs [hunk 1]
```diff
@@ -1565,6 +1565,7 @@ public static class UI
          public static readonly string ScaleButton = "Scale";
          public static readonly string AllowsScientificNotation = "Allows scientific notation";
          public static readonly string ShouldRestoreWorkspaceLayout = "Restore opened view when loading project";
+         public static readonly string ShouldColorGroupObservedData = "Color group observed data from same folder when dropping to chart";
```
Comment (msevestre) [inline src/PKSim.Assets/PKSimConstants.cs:1568]: You can define this in core. It should be the same for PKSim and MoBi right?

File: src/PKSim.Assets/PKSimConstants.cs [hunk 2]
```diff
@@ -1895,6 +1896,7 @@ public static class UI
          public static readonly string ComparisonViewDescription = "Show or hide the comparison";
          public static readonly string SimulationExplorerViewDescription = "Show or hide the simulation explorer";
          public static readonly string BuildingBlockExplorerViewDescription = "Show or hide the building block explorer";
+         public static readonly string ColorGroupObservedDataContextMenu = "Color group when adding to chart";
```
Comment (msevestre) [inline src/PKSim.Assets/PKSimConstants.cs:1899]: same with this

File: src/PKSim.Presentation/Presenters/Charts/ChartPresenter.cs [hunk 3]
```diff
@@ -51,7 +53,8 @@ public abstract class ChartPresenter<TChart, TView, TPresenter> : OSPSuite.Prese
          IIndividualPKAnalysisPresenter pkAnalysisPresenter,
          IChartTask chartTask, 
          IObservedDataTask observedDataTask, 
-         IChartUpdater chartUpdater, bool useSimulationNameToCreateCurveName)
+         IChartUpdater chartUpdater, bool useSimulationNameToCreateCurveName,
```
Comment (msevestre) [inline src/PKSim.Presentation/Presenters/Charts/ChartPresenter.cs:56]: align this parameter to the next line (old code but still, why not)

File: src/PKSim.UI/UserSettings.cs [hunk 4]
```diff
@@ -63,6 +63,7 @@ public class UserSettings : ValidatableDTO, IUserSettings
       public JournalPageEditorSettings JournalPageEditorSettings { get; set; }
       public string LastIgnoredVersion { get; set; }
       public OutputSelections OutputSelections { get; set; }
+      public bool ColorGroupObservedDataFromSameFolder { get; set; }
```
Comment (msevestre) [inline src/PKSim.UI/UserSettings.cs:66]: This needs to be added to serialization

**Review by msevestre — CHANGES_REQUESTED** [review]

File: src/PKSim.Presentation/Presenters/Charts/ChartPresenter.cs (hunk 3 above)
Comment (georgeDaskalakis) [reply src/PKSim.Presentation/Presenters/Charts/ChartPresenter.cs:56]: not sure how you mean... should "IUserSettings userSettings)" go to the line above?

File: src/PKSim.UI/UserSettings.cs (hunk 4 above)
Comment (georgeDaskalakis) [reply src/PKSim.UI/UserSettings.cs:66]: done

File: src/PKSim.Assets/PKSimConstants.cs (hunk 1 above)
Comment (georgeDaskalakis) [reply src/PKSim.Assets/PKSimConstants.cs:1568]: done

File: src/PKSim.Assets/PKSimConstants.cs (hunk 2 above)
Comment (georgeDaskalakis) [reply src/PKSim.Assets/PKSimConstants.cs:1899]: done

File: src/PKSim.Presentation/Presenters/Charts/ChartPresenter.cs (hunk 3 above)
Comment (msevestre) [reply src/PKSim.Presentation/Presenters/Charts/ChartPresenter.cs:56]: Lol no
The useSimulationNameToCreateCurve

But that's not a big deal.

**Comment by msevestre** [comment]
I can't merge this until you make a PR in core correct?

**Comment by georgeDaskalakis** [comment]
You can merge, it will just be merged in the feature branch here in PK-Sim. I have commited the changes in core (the constants move from here) directly to the feature branch. I mean since we have one, I think it is OK to do like that without a PR for jus the 2 lines of code for the definition right?

**Comment by msevestre** [comment]
>I think it is OK to do like that without a PR for jus the 2 lines of code for the definition right?
Oh yes for sure. I just did not see those changes that's why. Let's make sure we only do that for obvious trivial stuff. Adding constants etc.. is clearly the case.

**Pull request merged**